src/pages/HomePage.tsx

```diff
@@ -81,29 +82,105 @@ export class HomepageInternal extends React.Component<
+          </ScrollyItem>
+        );
+
+        // Add it to the array of packages to display
+        packagesToDisplay.push(pkgElement);
+      }
+    });
+
+    // There may be a chance that all the packages are old
+    if (packagesToDisplay.length > 0) {
       return (
         <ScrollyRow>
-          {this.state.displayedPackages.map(p => {
-            return (
-              <ScrollyItem key={p.id}>
-                <Link
-                  to={`/${this.props.context.selectedPackageSet!.slug}/${
-                    p.slug
-                  }`}
-                >
-                  <PackageCard package={p} />
-                </Link>
-              </ScrollyItem>
-            );
-          })}
+          {packagesToDisplay}
         </ScrollyRow>
       );
-    } else {
-      return <h2>Loading...</h2>;
     }
-  };
+    else return <p>No Packages Found</p>
+  }
+
+  // Renders N of the most recently updated packages
+  MAX_NUMBER_OF_RECENTLY_UPDATED_PACKAGES_TO_SHOW = 10;
+  renderRecentlyUpdatedPackages(): JSX.Element {
+    let packages = this.state.displayedPackages;
+    const MAX_NUMBER_OF_PACKAGES_TO_SHOW = this.MAX_NUMBER_OF_RECENTLY_UPDATED_PACKAGES_TO_SHOW;  // Can change this, default 10
+
+    // Straight return if there are no packages to display
+    if (!packages)
+      return <p>Loading...</p>;
+
+    // Sort packages by date - this will be very inefficient if there are many packages
+    packages = packages.sort((a: IPackage, b: IPackage): number => {
+      let aDate = new Date((a as any).updated_at), bDate = new Date((b as any).updated_at); // Type def doesn't have key 'updated_at', but it exists
+
+      if (aDate < bDate)      // aDate is before bDate, move to back
+        return 1;
+      else if (aDate > bDate) // aDate is after bDate, move to front
+        return -1;
+      else return 0;
+    });
+
+    // Display the first MAX_NUMBER_OF_PACKAGES_TO_SHOW packages of the sorted array
+    let packagesToDisplay: JSX.Element[] = [];
+    let numPackages = packages.length;
+    for (let i = 0; i < numPackages && i < MAX_NUMBER_OF_PACKAGES_TO_SHOW; i++) {
+      const pkg = packages[i];
+
+      // Create package element
+      let pkgElement = (
+        <ScrollyItem key={pkg.id}>
+          <Link to={`/${this.props.context.selectedPackageSet!.slug}/${pkg.slug}`}>
+            <PackageCard package={pkg} />
+          </Link>
+        </ScrollyItem>
+      );
+
+      packagesToDisplay.push(pkgElement);
+    }
+
+    // There may be a chance that all the packages are old
+    if (packagesToDisplay.length > 0) {
+      return (
+        <ScrollyRow>
+          {packagesToDisplay}
+        </ScrollyRow>
+      );
+    }
+    else return <p>No Recently Updated Packages</p>
+  }
+
+  changeMaxNumberOfRecentlyUpdatedPackages(e: ChangeEvent<HTMLSelectElement>) {
+    const dropdown = e.target as HTMLSelectElement;
+    this.MAX_NUMBER_OF_RECENTLY_UPDATED_PACKAGES_TO_SHOW = parseInt(dropdown.value);
```

Comment:
For this, I would recommend using state instead. Whenever you update state, it will trigger a re render.